pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -98,6 +97,52 @@ func Execute(configFile io.Reader) {
 		pruneAccessRecords,
 	)
 
+	// Registry extensions endpoint provides extra functionality to handle the image
+	// signatures.
+	extensionsRouter := app.NewRoute().PathPrefix("/extensions/v2/").Subrouter()
+	var (
+		getSignatureAccess = func(r *http.Request) []auth.Access {
+			return []auth.Access{
```

Comment:
Even better. Leave it as it is. But in the auth module, make sure to do the repository check just once.